exporter/collector/internal/datapointstorage/datapointcache.go

```diff
@@ -212,27 +212,57 @@ func (c *Cache) gc(shutdown <-chan struct{}, tickerCh <-chan time.Time) bool {
 }
 
 // Identifier returns the unique string identifier for a metric.
-func Identifier(resource *monitoredrespb.MonitoredResource, extraLabels map[string]string, metric pmetric.Metric, attributes pcommon.Map) string {
-	var b strings.Builder
+func Identifier(resource *monitoredrespb.MonitoredResource, extraLabels map[string]string, metric pmetric.Metric, attributes pcommon.Map) (uint64, error) {
+	var err error
+	h := fnv.New64()
 
-	// Resource identifiers
-	if resource != nil {
-		fmt.Fprintf(&b, "%v", resource.GetLabels())
+	_, err = h.Write([]byte(resource.GetType()))
+	if err != nil {
+		return 0, err
+	}
+	_, err = h.Write([]byte(metric.Name()))
+	if err != nil {
+		return 0, err
 	}
 
-	// Instrumentation library labels and additional resource labels
-	fmt.Fprintf(&b, " - %v", extraLabels)
-
-	// Metric identifiers
-	fmt.Fprintf(&b, " - %s -", metric.Name())
-	attrsIds := make([]string, 0, attributes.Len())
+	attrs := make(map[string]string)
 	attributes.Range(func(k string, v pcommon.Value) bool {
-		attrsIds = append(attrsIds, k+"="+v.AsString())
+		attrs[k] = v.AsString()
 		return true
 	})
-	if len(attrsIds) > 0 {
-		sort.Strings(attrsIds)
-		fmt.Fprint(&b, " "+strings.Join(attrsIds, " "))
+
+	err = hashOfMap(h, extraLabels)
+	if err != nil {
+		return 0, err
+	}
+
+	err = hashOfMap(h, attrs)
+	if err != nil {
+		return 0, err
+	}
+
+	err = hashOfMap(h, resource.GetLabels())
+	if err != nil {
+		return 0, err
+	}
+	return h.Sum64(), err
+}
+
+func hashOfMap(h hash.Hash64, m map[string]string) error {
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		_, err := h.Write([]byte(key))
```

Comment:
I would ignore all those errors explicitly `_ = ` for hash you use it's noop I believe (https://github.com/prometheus/prometheus/blob/main/model/labels/labels_dedupelabels.go#L191)

Reply:
updated to ignore errors